lang/mozjs78/Portfile

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim: fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           muniversal 1.0
+PortGroup           compiler_blacklist_versions 1.0
+
+name                mozjs78
+version             78.11.0
+categories          lang
+platforms           darwin
+license             {MPL-2 LGPL-2.1+}
+maintainers         {devans @dbevans} openmaintainer
+description         JavaScript-C Engine
+long_description    SpiderMonkey is Mozilla's JavaScript engine written in C/C++. \
+                    It is used in various Mozilla products, including Firefox, \
+                    and is available under the MPL2.
+
+homepage            https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey
+# build from GNOME releng tarball
+master_sites        https://download.gnome.org/teams/releng/tarballs-needing-help/mozjs/
+
+distname            mozjs-${version}
+use_bzip2           yes
+
+checksums           rmd160  9defcd91157c423f7af04a43291ae57a1755cf62 \
+                    sha256  5ad54ac2b0368b9748e08756658eb988d65e5bb088d91baf43267486bcbf1d26 \
+                    size    70531559
+
+depends_build       port:autoconf213 \
+                    port:pkgconfig \
+                    port:python27 \
+                    port:yasm \
+                    port:cargo \
+                    port:rust
+
+depends_lib         port:nspr \
+                    port:xorg-libX11 \
+                    port:xorg-libXt
+
+if { ${os.platform} eq "darwin" && ${os.major} < 11 } {
+    depends_build-append port:cctools
+    configure.env-append AR=${prefix}/bin/ar
+}
+
+patchfiles          patch-js-config.in.diff
+
+# Use absolute path for install_name
+post-patch {
+    reinplace "s|@executable_path|${prefix}/lib|g" ${worksrcpath}/config/rules.mk
+    reinplace "s|sdk_max_version = Version('10.15.4')|sdk_max_version = Version('12.1')|g" ${worksrcpath}/build/moz.configure/toolchain.configure
```

Comment:
Why is 12.1 the max SDK? What happens when the macOS 12.2 SDK is released?

Reply:
Here advice is really welcome. Officially, the max supported version of the SDK is 10.15.4 (if the installed SDK is more recent, the build scripts suggest downgrading to 10.15.4 using e.g. https://github.com/phracker/MacOSX-SDKs), and it turns out that building with the 12.1 SDK also succeeds (on Mac OS 12.1 at least), and build also succeeds under Mac OS 10.15 and 11 in the CI bots. Hence this new max SDK in the Portfile. What would be the best option?

Reply:
To be more precise, the build scripts of mozjs 78 suggest downloading an older version of the SDK, possibly in a local directory and, if needed, pointing at this directory thanks to ``https://github.com/phracker/MacOSX-SDKs'' in the mozconfig. If we don't want to build with the newer SDK (12.1 or later) installed system-wide on the machine where the build happens, this might be an option (download the older SDK in a temporary directory, build mozjs78, remove the SDK), but this means relying on a third party from which the binaries of the older SDK are downloaded.